### ISSUE TITLE
Social LSTM: Human Trajectory Prediction in Crowded Spaces

### ISSUE THREAD

**Issue by hsanchez** [hsanchez/paper-memory#6]
## TL;DR

![Fig. 1. Human Trajectories as seen by Social LSTMs](https://user-images.githubusercontent.com/199998/160007134-2757dd11-829b-4b23-8c1f-770570681e1e.png)
*Fig. 1. Human Trajectories as seen by Social LSTMs*

Social LSTMs is a DL model that predicts human trajectories given the data of their past trajectories (See Fig 1 for a high-level view). Social LSTMs’ architecture can be customized to support other kinds of objects’ trajectories.

### Paper Link

- [CVPR16_Social_LSTM paper](https://cvgl.stanford.edu/papers/CVPR16_Social_LSTM.pdf)
- [Social LSTMs model code](https://github.com/quancore/social-lstm)

### Author/Institution

- Alexandre Alahi (Stanford University)
- Kratarth Goel (Stanford University)
- Vignesh Ramanathan (Stanford University)
- Alexandre Robicquet (Stanford University)
- Li Fei-Fei (Stanford University)
- Silvio Savarese (Stanford University)

## Overview

Social LSTMs is a DL model that predicts human trajectories given the data of their past trajectories (See Fig 1 for a high-level view). Social LSTMs’ architecture can be customized to support other kinds of objects’ trajectories.

Social LSTMs model that can jointly reason across multiple individuals to predict human trajectories in a scene. We use one LSTM for each trajectory and share the information between the LSTMs through the introduction of a new Social pooling layer. Each scene is first pre-processed to obtain the spatial coordinates of all people at different time instants. A scene is a sequence of frames from a top-view static camera. At any time-instant, the $$i_{th}$$ person in the scene is represented by their $$(x,y)$$ coordinates $$(x^{i}_{t},y^{i}_{t})$$. We observed the positions of all the people from time $$1$$ to $$T_{obs}$$ (input sequence), and predict their positions for time-instants $$T_{obs} + 1$$ to $$T_{pred}$$ (future predicted sequence).

![social_lstms_overview](https://user-images.githubusercontent.com/199998/160028663-23a6d164-7b07-4e6c-aa02-7e5fa6d599d1.png)
*Fig. 2. Overview of the Social-LSTM model.  There is a separate LSTM network for each trajectory in a scene. The LSTMs are then connected to each other through a Social pooling (S-pooling) layer. Unlike the traditional LSTM, this pooling layer allows spatially proximal LSTMs to share information with each other. $$h_{x}$$ represents the hidden state of the LSTM for person $$x$$. The bottom row shows the S-pooling for one person in the scene. The hidden states of all LSTMs within a certain radius are pooled together and used as input.*

The LSTM weights are shared across all the sequences.

Social LSTMs outperform state-of-the-art methods on two publicly available datasets. It is successful at predicting various non-linear behaviors arising from social interactions, such as a group of individuals moving together.

Future work will extend our model to multi-class settings where several objects such as bicycles, skateboards, carts, and pedestrians share the same space.  Each object will have its own label on the occupancy map. In addition, human-space interaction can be modeled in our framework by including the local static-scene image as an additional input to the LSTM. This could allow jointly modeling of human-human and human-space interactions in the same framework.

## Contributions and Distinctions from Previous Works

Earlier works in human trajectory prediction are limited by the following:

1. Use of hand-crafted functions to model “interactions” for specific settings. This results in models that capture simple interactions (e.g. repulsion/attractions) and that might fail to generalize for more complex crowded settings.
2. Modeling of interactions among people in close proximity (w.r.t. time) to each other. Interactions that could occur in the more distant future are not covered.

This is where the Social LSTMs learning architecture fills the void through "Social" pooling.

### Social Pooling of hidden states

In a crowded scene, individuals would adjust their paths by implicitly reasoning about the motion of nearby individuals. In turn, the paths of neighboring individuals are influenced by others in their immediate surroundings and could alter their behavior over time. Social LSTMs expect the hidden states of an LSTM to capture these time-varying motion properties. In order to jointly reason across multiple people, the model introduces “Social” pooling layers as shown in Fig. 2. *At every time step, the LSTM cell receives pooled hidden-state information from the LSTM cells of neighbors.*

![social_pooling](https://user-images.githubusercontent.com/199998/160029273-2b4a04b0-9124-4fa9-a72f-24f71b9a7487.png)
*Fig. 3. The figure shows the Social pooling for the person represented by a black dot. Model pools the hidden states of the neighbors (shown as in yellow, blue, and orange, represented by $$h$$ values) within a certain spatial distance. The pooling partially preserves the spatial info of neighbors as shown in the last two steps.*

The hidden state at a time $$t$$ is used to predict the distribution of the trajectory position $$(x',y')_{t + 1}$$ at the next time-step $$t + 1$$.

## Results

![performance_results](https://user-images.githubusercontent.com/199998/160029886-a7c8d260-3393-4b72-823d-ffe90142debd.png)
*Performance of Social LSTM compared to the other state-of-the-art trajectory prediction models. Further details are in the 'Results' section*
